runtime/integration-tests/src/test_macros.rs

```diff
@@ -0,0 +1,468 @@
+		$network_id: ident
+	) => {{
+		use crate::mock::{BOB, TEN_UNITS, UNIT};
+		use frame_support::traits::{fungible::Mutate, fungibles::Inspect};
+		use polkadot_core_primitives::AccountId;
+		use xcm::latest::{
+			Junction, Junction::Parachain, Junctions::X2, MultiLocation, WeightLimit,
+		};
+
+		//first we need to set up USDT balance on pendulum chain before to start transfer it back.
+		parachain1_transfer_asset_to_parachain2!(
+			$para1_runtime,
+			$parachain1,
+			$para1_asset_id,
+			$para2_runtime,
+			$parachain2,
+			$parachain2_id
+		);
+
+		$parachain1::execute_with(|| {});
+
+		$parachain2::execute_with(|| {
+			use $para2_runtime::{
+				Balances, CurrencyId, RuntimeEvent, RuntimeOrigin, System, Tokens, XTokens,
+			};
+
+			assert_eq!(TEN_UNITS, Tokens::balance(CurrencyId::XCM(1), &AccountId::from(BOB)));
+			// ensure sender has enough balance to be charged as fee
+			assert_ok!(Balances::mint_into(&AccountId::from(BOB), TEN_UNITS));
+
+			assert_ok!(XTokens::transfer(
+				RuntimeOrigin::signed(BOB.into()),
+				CurrencyId::XCM(1),
+				UNIT * 1,
+				Box::new(
+					MultiLocation::new(
+						1,
+						X2(
+							Parachain($parachain1_id),
+							Junction::AccountId32 { network: Some($network_id), id: BOB.into() }
+						)
+					)
+					.into()
+				),
+				WeightLimit::Unlimited
+			));
+
+			assert_eq!(
+				TEN_UNITS - 1 * UNIT, //initial balance - one unit
+				Tokens::balance(CurrencyId::XCM(1), &AccountId::from(BOB))
+			);
+
+			assert!(System::events().iter().any(|r| matches!(
+				r.event,
+				RuntimeEvent::XcmpQueue(cumulus_pallet_xcmp_queue::Event::XcmpMessageSent { .. })
+			)));
+
+			for i in System::events().iter() {
+				println!("{}: {:?}\n", stringify!($para2_runtime), i);
+			}
+		});
+
+		$parachain1::execute_with(|| {
+			use $para1_runtime::*;
+
+			for i in System::events().iter() {
+				println!("{}: {:?}\n", stringify!($para1_runtime), i);
+			}
+
+			let events = System::events();
+			match &events[events.len() - 2] {
+				&frame_system::EventRecord {
+					phase: frame_system::Phase::Initialization,
+					event:
+						RuntimeEvent::Assets(pallet_assets::Event::Issued {
+							asset_id: $para1_asset_id,
+							owner: _,
+							amount,
+						}),
+					topics: _,
+				} => {
+					// https://github.com/paritytech/cumulus/pull/1278 support using self sufficient asset
+					// for paying xcm execution fee on Statemint.
+					// 990_000_000_000 for Statemint
+					// 988_423_297_485 for Statemine
+					assert_eq!(amount, Assets::balance($para1_asset_id, &AccountId::from(BOB)));
+				},
+				other => panic!("wrong event: {other:?}"),
+			}
+		});
+	}};
+}
+
+pub(super) use parachain1_transfer_asset_to_parachain2;
+pub(super) use parachain1_transfer_asset_to_parachain2_and_back;
+pub(super) use parachain1_transfer_incorrect_asset_to_parachain2_should_fail;
+pub(super) use transfer_10_relay_token_from_parachain_to_relay_chain;
+pub(super) use transfer_20_relay_token_from_relay_chain_to_parachain;
```

Comment:
Should we move this to the top of the file maybe?

Reply:
i tried it before, and it resulted to `unresolved import` and `cannot determine resolution for the macro...`

Reply:
Oh, okay 😅 Maybe we should add a little comment then that these are intentionally defined at the bottom of the file and we encounter that error otherwise?

Reply:
done,